### PR TITLE
[KED-2633] Add split panel component and implement into sidebar

### PR DESCRIPTION
**Pull request by bru5**
## Description

Adds a new split panel component and splits the current sidebar into two sections.

split-panel-2

## Development notes

There is a new generic `SplitPanel` component, which implements user resizing.

The existing sidebar has been split into two panels, the elements section is now in the top and the categories are in the bottom panel.

This work is the first part of a series of design changes coming to the sidebar.

## QA notes

Other than the split panel, functionality is the same as existing, including all existing search and filters which should be tested through and compared to the previous release.

Usability should also be tested, focus and keyboard arrows are implemented.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: package/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 semver~=2.10 # Needs to be at least 2.10.0 to get VersionInfo.match
-Flask>=1.0, <2.0
+Flask>=2.0
```
Comment (studioswong) [inline package/requirements.txt:2]: Is this change related to / required by this PR?

Comment (studioswong) [reply package/requirements.txt:2]: Just saw the commit history and it looks like this has been updated yesterday from lim - perhaps this should be introduced in a seperate PR?

File: src/components/split-panel/index.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+}) => {
+  const containerRef = useRef();
+  const handleRef = useRef();
+
+  const getRects = () => ({
+    container: containerRef.current?.getBoundingClientRect(),
+    handle: handleRef.current?.getBoundingClientRect(),
+  });
+
+  const clampSplit = (value) => {
```
Comment (studioswong) [inline src/components/split-panel/index.js:29]: I know it's hard to find a good variable name for this, but `clampSplit` sounds a bit confusing - perhaps calling it 'calculateSplit` or `calculateXX` ( whatever it is trying to calculate) would be more intuitive?

File: src/components/split-panel/index.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+/**
+ * A design agnostic user resizable split panel controller.
+ * @param {function} children A `function(props)` for rendering the panels and handle (see implementation for props)
+ * @param {?number} [splitDefault=0.65] A number [0...1] as the default % split position
+ * @param {?number} [splitMin=0] A number [0...1] as the minimum % split position
+ * @param {?number} [splitMax=1] A number [0...1] as the maximum % split position
+ * @param {?number} [keyboardStep=0.025] A number [0...1] as the % step to move split when using keyboard
+ * @param {?string} [orientation='vertical'] Only 'vertical' currently supported.
+ * @return {object} The rendered children
```
Comment (studioswong) [inline src/components/split-panel/index.js:11]: It would be more clear to add some example of what the children usually is and any expected format of the children in the comment, such as the react component it is encapsulating, etc

**Review by studioswong — APPROVED** [review]
Really great work @bru5 , well done 👍

**Review by rashidakanchwala — APPROVED** [review]

File: src/components/split-panel/index.js (hunk 2 above)
Comment (bru5) [reply src/components/split-panel/index.js]: Fair enough if `clamp` seems to clash with `split` I guess, I've refactored this to just `clamp` then.

File: package/requirements.txt (hunk 1 above)
Comment (bru5) [reply package/requirements.txt]: Yeah this was a temporary fix by Lim, I think this is resolved in the main branch now though which I've merged here.

File: src/components/split-panel/index.js (hunk 3 above)
Comment (bru5) [reply src/components/split-panel/index.js]: Sure I've improved the documentation in this file, including a note that the test file serves as a reference usage.
